rust/arrow/src/csv/reader.rs

```diff
@@ -633,6 +636,45 @@ fn build_primitive_array<T: ArrowPrimitiveType + Parser>(
         .map(|e| Arc::new(e) as ArrayRef)
 }
 
+// parses a specific column (col_idx) into an Arrow Array.
+fn build_decimal_array(
+    line_number: usize,
+    rows: &[StringRecord],
+    col_idx: usize,
+    precision: usize,
+    scale: usize,
+) -> Result<ArrayRef> {
+    let mut builder = DecimalBuilder::new(rows.len(), precision, scale);
+
+    for (row_index, row) in rows.iter().enumerate() {
+        match row.get(col_idx) {
+            Some(s) => {
+                if s.is_empty() {
+                    builder.append_null()?
+                }
+
+                let parsed = match Decimal128Type::parse(s, precision, scale) {
+                    Ok(number) => number,
+                    Err(_) => {
+                        return Err(ArrowError::ParseError(format!(
+                            // TODO: we should surface the underlying error here.
+                            "Error while parsing value {} for column {} at line {}",
+                            s,
+                            col_idx,
+                            line_number + row_index
```

Comment:
```suggestion
                            "Error while parsing value {} for column {} at line {}: {}",
                            s,
                            col_idx,
                            line_number + row_index,
                            e
```

might be one way of surfacing the underlying error